src/sidebar/services/session.js

```diff
@@ -10,31 +10,32 @@ const CACHE_TTL = 5 * 60 * 1000; // 5 minutes
  * This service handles fetching the user's profile, updating profile settings
  * and logging out.
  *
- * Access to the current profile is exposed via the `state` property.
- *
- * @param {import('../store').SidebarStore} store
- * @param {import('./api').APIService} api
- * @param {import('./auth').AuthService} auth
- * @param {import('./toast-messenger').ToastMessengerService} toastMessenger
  * @inject
  */
-export default function session(store, api, auth, settings, toastMessenger) {
-  // Cache the result of load()
-  let lastLoad;
-  let lastLoadTime;
-
-  // Return the authority from the first service defined in the settings.
-  // Return null if there are no services defined in the settings.
-  function getAuthority() {
-    const service = serviceConfig(settings);
-    if (service === null) {
-      return null;
-    }
-    return service.authority;
-  }
+export class SessionService {
+  /**
+   * @param {import('../store').SidebarStore} store
+   * @param {import('./api').APIService} api
+   * @param {import('./auth').AuthService} auth
+   * @param {import('./toast-messenger').ToastMessengerService} toastMessenger
```

Comment:
looks like `settings` is missing from the param types

Reply:
That's currently true of all services. My plan was to go around and address that in all services separately.